### PR TITLE
Add helpful cli app

### PR DESCRIPTION
**Pull request by lilatomic**
adds a helpful cli app to template Dashboards and Alerts!

### CODE REVIEW

File: src/grafanarmadillo/cmd.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"The type should be a mapping of environment names (like 'template' or 'production') " + \
+	"to dicts of replacements (for example, `{\"region\": \"SouthWest\"}`). " + \
+	load_file_help
+env_grafana_help = "Name of the environment in the mapping file for Grafana, found in the `--mapping` argument"
+env_template_help = "Name of the environment in the mapping file for the template, found in the `--mapping` argument"
+
+
+def load_data(data_str: str):
+	"""Attempt to load data."""
+	_file_uri_prefix = "file://"
```
Comment (simonsben) [inline src/grafanarmadillo/cmd.py:30]: Could probably live at the module level as a contant

File: src/grafanarmadillo/cmd.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	if data_str.startswith(_file_uri_prefix):
+		filename = Path(data_str.split(_file_uri_prefix)[1])
```
Comment (simonsben) [inline src/grafanarmadillo/cmd.py:32]: I really don't like the split approach for this stuff, shame it's not `3.9+`, cause then you could just throw in a `str.removeprefix`.. almost worth using something like `data_str[len(_file_uri_prefix):]`, but that's also terrible

File: src/grafanarmadillo/cmd.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		return json.loads(data_str)
+
+
+def make_mapping_templator(mapping: EnvMapping, env_grafana: str, env_template: str) -> Templator:
+	"""Assemble the templator from the environment mapping."""
+	mapping_grafana = mapping[env_grafana]
+	mapping_template = mapping[env_template]
+
+	# if some keys in the src mapping are not in the dst mapping
+	missing = set(mapping_grafana.keys()) - set(mapping_template.keys())
```
Comment (simonsben) [inline src/grafanarmadillo/cmd.py:45]: I don't think you need to cast to set explicitly, i think the key set is a child?

File: src/grafanarmadillo/cmd.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	if missing:
+		raise ValueError(f"Some keys in the source mapping are not present in the destination mapping. {missing=}")
+
+	grafana_to_template = {}
+	for k, v in mapping_grafana.items():
+		grafana_to_template[v] = mapping_template[k]
```
Comment (simonsben) [inline src/grafanarmadillo/cmd.py:51]:
```suggestion
	grafana_to_template = {value: mapping_template[key] for key, value in mapping_grafana.items()}
```
worth just throwing as a one-liner?

File: src/grafanarmadillo/find.py [hunk 5]
```diff
@@ -53,12 +58,11 @@ def get_folder(self, name) -> FolderSearchResult:
...
-					filter(
-						lambda x: x["title"] == name,
-						map(lambda sr: self.api.folder.get_folder(sr["uid"]), search_result),
-					)
-				)
+				list(filter(
+					lambda x: x["title"] == name,
+					map(lambda sr: self.api.folder.get_folder(sr["uid"]), search_result),
+				)),
+				_query_message("folder", name),
```
Comment (simonsben) [inline src/grafanarmadillo/find.py:65]: Feel like i've been leaning away from `filter` seems like you loose some linting/cleanliness from them, any sense is splitting it into a couple steps, as in,
```python
folders = (self.api.folder.get_folder(entry["uid"]) for entry in search_result)
return exactly_one(
  [entry for entry in folders if entry["title"] == name],
  _query_message("folder", name),
)
```

File: tests/cli/test_cli.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+			"--env-template",
+			"template",
+			"--mapping",
+			"file://tests/cli/mapping.json",
+		]
+	)
+	assert result.exit_code == 0
+
+	with template_path.open(mode="r") as template_file:
+		template = json.loads(template_file.read())
```
Comment (simonsben) [inline tests/cli/test_cli.py:84]: Might be worth adding a central `load_file` type method with json load support?

**Review by simonsben — COMMENTED** [review]
idk if you were looking for thoughts, but generally
* sad that it's only `3.6+`, some newer builtins would be nice
* tests could use some more type hints imo
* def a style decision, but feel like filter/map with lambdas don't provide the same rename and type hint safety that a comprehension does

File: src/grafanarmadillo/cmd.py (hunk 2 above)
Comment (lilatomic) [reply src/grafanarmadillo/cmd.py:32]: yeah I should probably just urlparse it

File: src/grafanarmadillo/cmd.py (hunk 3 above)
Comment (lilatomic) [reply src/grafanarmadillo/cmd.py:45]: oh neat, I didn't know that dict_keys was setlike.

**Pull request merged**